packages/patrol/lib/src/custom_finders/patrol_tester.dart

```diff
@@ -63,15 +65,23 @@ class PatrolTester {
   /// Flutter's widget tester that this [PatrolTester] wraps.
   final WidgetTester tester;
 
-  /// Native automator that allows for interaction with the host OS.
+  /// Native automator that allows for interaction with OS the app is running on.
   final NativeAutomator? nativeAutomator;
 
+  /// Provides functionality to run actions as the host system (your computer).
+  final HostAutomator? hostAutomator;
+
   /// Shorthand for [nativeAutomator]. Throws if [nativeAutomator] is null.
   NativeAutomator get native {
     assert(nativeAutomator != null, 'native automator is null');
     return nativeAutomator!;
   }
 
+  HostAutomator get host {
+    assert(hostAutomator != null, 'host automator is null');
```

Comment:
Usually assert messages say what was expected, explain the needed invariants (for instance `'host getter can be used only if hostAutomator was initialized before'` or something). Because currently the error message is not helpful.

Reply:
Thanks, I agree.

Reply:
In the end, I improved the doc comment above `native` and `host` fields instead of improving the assertion message. 

Albert told me some time ago that assertions should be in the form of:

`X is null`, `Y is not in the inclusive 1..10 range`

instead of:

`X must not be null`, `Y must be in the inclusive 1..10 range`

Reply:
> Albert told me some time ago that assertions should be in the form of:
> 
> `X is null`, `Y is not in the inclusive 1..10 range`
> 
> instead of:
> 
> `X must not be null`, `Y must be in the inclusive 1..10 range`

I cannot say I agree... But even then, I don't think such error messages are helpful at all. They don't give any meaningful information. I know what the value should or should not be, I can see what you are asserting (dart's `assert` is not a function, it is a language feature, they capture the expression and print the exact expression that was used to perform the assert, see attached screenshot). Asserts are for _enforcing invariants_ so the error message should explain what kind of invariant was not fulfilled. Moving this to docs is great, though.

<img width="929" alt="image" src="https://user-images.githubusercontent.com/29288116/201763418-bfe7e4dd-0fa0-4ea3-ab46-9d7a22a84d16.png">

Coming back to this specific case, as an outside person who read the docs above this getter I still have no idea what should I do if I encounter a null here. Why would it be not initialized? Is it allowed to be not initialized? It is a `final` value, so I don't think I can initialize it myself.